custom_components/opnsense/__init__.py

```
@@ -159,18 +160,37 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+            hass,
+            DOMAIN,
+            f"{config_device_id}_missing_external_aiopnsense",
+            is_fixable=False,
+            is_persistent=False,
+            issue_domain=DOMAIN,
+            severity=ir.IssueSeverity.ERROR,
+            translation_key="missing_external_aiopnsense",
+        )
+        _LOGGER.error(
+            "OPNsense firmware requires external aiopnsense package, but it is not available."
+        )
```

Ruff (0.15.7)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)